OneSignalExample/Assets/OneSignal/Platforms/iOS/OneSignalUnityRuntime.m

```diff
@@ -244,38 +244,74 @@ void _deleteTags(const char* keys) {
         [OneSignal deleteTags:kk];
 }
 
-void _getTags() {
+void _getTags(const char* delegate) {
+    NSString* delegateId = @"";
+    if (delegate)
+        delegateId = CreateNSString(delegate);
+    
     [OneSignal getTags:^(NSDictionary* result) {
-        UnitySendMessage(unityListener, "onTagsReceived", dictionaryToJsonChar(result));
+        if (!result)
+            result = @{};
+        
+        NSString* response = dictionaryToNSString(result);
+        NSDictionary *data = @{ @"delegate_id" : delegateId, @"response" : response };
+        
+        UnitySendMessage(unityListener, "onTagsReceived", dictionaryToJsonChar(data));
     }];
 }
 
-void _idsAvailable() {
+void _idsAvailable(const char* delegate) {
+    NSString* delegateId = @"";
+    if (delegate)
+        delegateId = CreateNSString(delegate);
+    
     [OneSignal IdsAvailable:^(NSString* userId, NSString* pushToken) {
+        if (!userId)
+            userId = @"";
+            
         if (!pushToken)
             pushToken = @"";
         
-        UnitySendMessage(unityListener, "onIdsAvailable",
-                         dictionaryToJsonChar(@{@"userId" : userId, @"pushToken" : pushToken}));
+        NSString* response = dictionaryToNSString(@{ @"userId" : userId, @"pushToken" : pushToken });
+        NSDictionary* data = @{ @"delegate_id" : delegateId, @"response" : response };
+        
+        UnitySendMessage(unityListener, "onIdsAvailable", dictionaryToJsonChar(data));
     }];
 }
 
 void _setSubscription(BOOL enable) {
     [OneSignal setSubscription:enable];
 }
 
-void _postNotification(const char* jsonData) {
+void _postNotification(const char* delegateSuccess, const char* delegateFailure, const char* jsonData) {
+    NSString* delegateIdSuccess = @"";
+    if (delegateSuccess)

```

Comment:
Could probably clean up all these `null` checks if we have `CreateNSString` create us a blank `NSString`.